services/user.js

```diff
@@ -200,14 +201,22 @@ class UserService {
 
     async getUser(params, context) {
         verifySession(context)
-            .verifyInitialized()
-            .verifyPermission(USER_PERMISSION_CONSTANTS.ADMIN.MANAGE_USER);
-        
+            .verifyInitialized();
+        await this.#isPermitted(context?.userInfo, USER_PERMISSION_CONSTANTS.ADMIN.MANAGE_USER);
+
         if (!params?.userID) {
             throw new Error(SUBMODULE_ERROR.INVALID_USERID);
         }
+        const isFederalLeadOnly = await this.#isManageUserScope(context?.userInfo, ROLES.FEDERAL_LEAD);
```

Comment:
We should not check if the allowed role is Federal Lead at all, because it might be other roles in the future. Instead, we just need to know if the scope is "role" or "all", then use this information to create filters.